### PR TITLE
feat: pipeLazy (#218)

### PR DESCRIPTION
**Pull request by darky**

### CODE REVIEW

File: src/Lazy/pipeLazy.ts [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+ * // with toAsync
+ * await pipeLazy(
+ *  toAsync,
+ *  map(a => a + 10),
+ *  filter(a => a % 2 === 0),
+ *  toArray,
+ * )([Promise.resolve(1), Promise.resolve(2), Promise.resolve(3), Promise.resolve(4), Promise.resolve(5)]); // [12, 14]
+ * ```
+ *
+ * {@link https://codesandbox.io/s/fxts-toarray-fy84i | Try It}
```
Comment (darky) [inline src/Lazy/pipeLazy.ts:41]: @ppeeou @shine1594 can you please fix/correct only this part (codesandbox, doc, doc link)

Comment (ppeeou) [reply src/Lazy/pipeLazy.ts:41]: sure :)

Could you add type test? 
you can see reference https://github.com/marpple/FxTS/tree/main/type-check

Comment (darky) [reply src/Lazy/pipeLazy.ts:41]: > Could you add type test? you can see reference https://github.com/marpple/FxTS/tree/main/type-check

✅

Comment (darky) [reply src/Lazy/pipeLazy.ts:41]: @ppeeou if you have another notices, I can fix it too

**Comment by ppeeou** [comment]
@darky Thank you for your contribution
sorry for the late confirmation

I left a comment
In the following situation, function inference is wrong
Is it possible to implement a function to be inferred as a 'Promise' ?

```ts
const res = pipeLazy((a: number) => String(a))(Promise.resolve(2));
// infer res -> String not Promise(String)
```

**Comment by darky** [comment]
> I left a comment In the following situation, function inference is wrong Is it possible to implement a function to be inferred as a 'Promise' ?
> 
> ```ts
> const res = pipeLazy((a: number) => String(a))(Promise.resolve(2));
> // infer res -> String not Promise(String)
> ```

@ppeeou fixed

**Review by ppeeou — APPROVED** [review]
LGTM👍

**Pull request merged**